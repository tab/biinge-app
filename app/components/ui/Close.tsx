import React from "react"
import { Pressable, StyleProp, ViewStyle } from "react-native"
import Animated, { FadeIn } from "react-native-reanimated"
import { useTheme, useNavigation } from "@react-navigation/native"

import colors from "styles/colors"
import Icon from "components/ui/Icon"
import { closeStyles, layoutStyles } from "styles"

type Props = {
  style?: StyleProp<ViewStyle>
}

const CloseComponent = ({ style }: Props) => {
  const navigation = useNavigation()
  const { dark } = useTheme()

  const handleClose = () => {
    navigation.goBack()
  }

  return (
    <Pressable style={[closeStyles.root, style]} onPress={handleClose}>
      <Animated.View
        entering={FadeIn.delay(500)}
        style={[
          closeStyles.content,
          dark ? layoutStyles.bgDark : layoutStyles.bgLight,
        ]}
      >
        <Icon
          name="close-outline"
          color={dark ? colors.white : colors.graniteGray}
          size={21}
        />
      </Animated.View>
    </Pressable>
  )
}

export default CloseComponent
