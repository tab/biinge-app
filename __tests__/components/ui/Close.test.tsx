import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { NavigationContainer } from "@react-navigation/native"

import { it } from "@jest/globals"

import Close from "components/ui/Close"

describe("Close", () => {
  const handleClose = jest.fn()

  it("renders correctly", () => {
    const tree = render(
      <NavigationContainer>
        <Close onPress={handleClose} />
      </NavigationContainer>,
    )
    expect(tree).toMatchSnapshot()
  })
})
