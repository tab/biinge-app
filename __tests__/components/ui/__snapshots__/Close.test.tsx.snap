// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "left": 0,
        "padding": 12,
        "position": "absolute",
        "top": 0,
        "zIndex": 10,
      },
      undefined,
    ]
  }
>
  <View
    entering="FadeIn"
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 50,
          "height": 28,
          "justifyContent": "center",
          "width": 28,
        },
        {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#666666",
            "fontSize": 21,
          },
          undefined,
          {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      Ó≠è
    </Text>
  </View>
</View>
`;
